src/Stryker.Core/Stryker.Core/DiffProviders/GitDiffProvider.cs

```diff
@@ -1,81 +1,116 @@
+using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.IO;
 using System.Linq;
 using LibGit2Sharp;
+using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Extensions;
+using Spectre.Console;
 using Stryker.Abstractions.Exceptions;
 using Stryker.Abstractions.Options;
 using Stryker.Core.Baseline.Providers;
 using Stryker.Core.Mutants;
 using Stryker.Utilities;
 
-namespace Stryker.Core.DiffProviders;
-
-public class GitDiffProvider : IDiffProvider
+namespace Stryker.Core.DiffProviders
 {
-    public TestSet Tests { get; }
-    private readonly IStrykerOptions _options;
-    private readonly IGitInfoProvider _gitInfoProvider;
-
-    public GitDiffProvider(IStrykerOptions options, TestSet tests, IGitInfoProvider gitInfoProvider = null)
+    public class GitDiffProvider : IDiffProvider
     {
-        Tests = tests;
-        _options = options;
-        _gitInfoProvider = gitInfoProvider ?? new GitInfoProvider(options);
-    }
-
-    public DiffResult ScanDiff()
-    {
-        var diffResult = new DiffResult()
-        {
-            ChangedSourceFiles = new Collection<string>(),
-            ChangedTestFiles = new Collection<string>()
-        };
-
-        // A git repository has been detected, calculate the diff to filter
-        var repository = _gitInfoProvider.Repository;
-        var commit = _gitInfoProvider.DetermineCommit();
+        public TestSet Tests { get; }
```

Comment:
Why did we move these inside class?

Reply:
@inishantmishra We did not change anything there it was like that only